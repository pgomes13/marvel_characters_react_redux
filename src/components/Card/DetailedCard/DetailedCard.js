import React from 'react';
import PropTypes from 'prop-types';
import { withStyles } from '@material-ui/core/styles';
import CardMedia from "@material-ui/core/CardMedia/CardMedia";
import Card from "@material-ui/core/Card/Card";
import Typography from "@material-ui/core/Typography/Typography";
import CardContent from "@material-ui/core/CardContent/CardContent";

import { DetailedCardStyles } from './DetailedCard.styles';
import Hidden from "@material-ui/core/Hidden/Hidden";

/**
 * The DetailedCard component consisting of material-ui Card component
 * @param {Object} classes - the material-ui classes prop
 * @param {String} imageUrl - the image url prop
 * @param {String} title - the title prop
 * @param {String} description - the description prop
 * @returns {Node} - the DetailedCard component
 */
const _DetailedCard = ({ classes, imageUrl, title, description }) => (
    <Card className={classes.card}>
        <div className={classes.cardDetails}>
            <CardContent>
                <Typography component="h2" variant="h5">
                    { title }
                </Typography>
                <Typography variant="subtitle1" color="textSecondary">
                    { description }
                </Typography>
            </CardContent>
        </div>
        <Hidden xsDown>
            <CardMedia
                className={classes.cardMedia}
                image={imageUrl}
                title="Image title"
            />
        </Hidden>
    </Card>
);

_DetailedCard.propTypes = {
    classes: PropTypes.object.isRequired,
    imageUrl: PropTypes.string,
    title: PropTypes.string,
    description: PropTypes.string
};

const DetailedCard = withStyles(DetailedCardStyles)(_DetailedCard);

export { _DetailedCard, DetailedCard };
