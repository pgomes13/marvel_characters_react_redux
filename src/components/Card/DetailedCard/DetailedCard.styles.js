const DetailedCardStyles = theme => ({
    card: {
        display: 'flex',
        height: 200
    },
    cardMedia: {
        width: 200,
    },
    cardDetails: {
        flex: 1,
    }
});

export { DetailedCardStyles };