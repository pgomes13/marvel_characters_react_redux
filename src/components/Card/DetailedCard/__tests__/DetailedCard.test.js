import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import { _DetailedCard } from '../DetailedCard';

describe('DetailedCard', () => {
    const props = {
        classes: {},
        imageUrl: 'http://test.test.com',
        title: 'test',
        description: 'test'
    };

    it('component renders correctly with props', () => {
        const wrapper = shallow(<_DetailedCard {...props} />);
        expect(toJson(wrapper)).toMatchSnapshot();
    });
});