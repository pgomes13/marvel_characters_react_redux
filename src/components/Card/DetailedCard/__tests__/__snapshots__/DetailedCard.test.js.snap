// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DetailedCard component renders correctly with props 1`] = `
<WithStyles(Card)>
  <div>
    <WithStyles(CardContent)>
      <WithStyles(Typography)
        component="h2"
        variant="h5"
      >
        test
      </WithStyles(Typography)>
      <WithStyles(Typography)
        color="textSecondary"
        variant="subtitle1"
      >
        test
      </WithStyles(Typography)>
    </WithStyles(CardContent)>
  </div>
  <Hidden
    implementation="js"
    lgDown={false}
    lgUp={false}
    mdDown={false}
    mdUp={false}
    smDown={false}
    smUp={false}
    xlDown={false}
    xlUp={false}
    xsDown={true}
    xsUp={false}
  >
    <WithStyles(CardMedia)
      image="http://test.test.com"
      title="Image title"
    />
  </Hidden>
</WithStyles(Card)>
`;
